applications/adv_networking_bench/cpp/testing/benchmark_utils.py

```diff
@@ -0,0 +1,415 @@
+                    logger.info(
+                        f"Missed packets: {missed_rx_pkts} ({missed_pkts_percent:.2f}%, below {threshold}% threshold ✅)"
+                    )
+
+        return success
+
+    def validate_errored_packets(self, port_map: Dict[int, int], threshold: float) -> bool:
+        """
+        Validate errored packets are below threshold.
+
+        Args:
+            port_map: Dictionary mapping TX ports to RX ports (e.g., {0: 1})
+            threshold: Maximum allowed percentage of errored packets
+
+        Returns:
+            bool: True if validation passed, False otherwise
+        """
+        success = True
+
+        for tx_port, rx_port in port_map.items():
+            errored_rx_pkts = self.get_errored_packets(rx_port)
+            tx_pkts = self.get_tx_packets(tx_port)
+
+            if tx_pkts > 0:
+                errored_pkts_percent = (errored_rx_pkts / tx_pkts) * 100
+                if errored_pkts_percent > threshold:
+                    logger.error(
+                        f"Errored packets: {errored_rx_pkts} ({errored_pkts_percent:.2f}%, over {threshold}% threshold ❌)"
+                    )
+                    success = False
+                else:
+                    logger.info(
+                        f"Errored packets: {errored_rx_pkts} ({errored_pkts_percent:.2f}%, below {threshold}% threshold ✅)"
+                    )
+
+        return success
+
+    def validate_throughput(self, port_map: Dict[int, int], threshold: float) -> bool:
+        """
+        Validate throughput is above threshold.
+
+        Args:
+            port_map: Dictionary mapping TX ports to RX ports (e.g., {0: 1})
+            threshold: Minimum required throughput in Gbps
+
+        Returns:
+            bool: True if validation passed, False otherwise
+        """
+        success = True
+
+        for tx_port, rx_port in port_map.items():
+            rx_throughput = self.get_rx_throughput(rx_port)
+
+            if rx_throughput < threshold:
+                logger.error(
+                    f"Average throughput: {rx_throughput:.2f} Gbps (below {threshold} Gbps threshold ❌)"
+                )
+                success = False
+            else:
+                logger.info(
+                    f"Average throughput: {rx_throughput:.2f} Gbps (above {threshold} Gbps threshold ✅)"
+                )
+
+        return success
+
+
+def parse_benchmark_results(log: str) -> BenchmarkResults:
+    """
+    Parse benchmark results from log output.
+
+    Args:
+        log: The log output as a string
+
+    Returns:
+        BenchmarkResults: A structured representation of the benchmark results
+    """
+    # Initialize result dictionaries
+    tx_packets = {}
+    tx_bytes = {}
+    rx_packets = {}
+    rx_bytes = {}
+    missed_packets = {}
+    errored_packets = {}
+    rx_queue_packets = {}
+    tx_queue_packets = {}
+
+    # Regex patterns
+    port_pattern = r"Port (\d+):"
+    metric_patterns = {
```

Comment:
Should this be specialized for DPDK or should we instead specialize it for the common stats manager output that they all can use?

Reply:
Currently looks like this parser is specialized for DPDK, because that's all we have.

two options:
1. add separate parsers for other backends once we start using them
2. provide a common stats logging mechanism that is consistent across managers, and have a single parser for that


Either of these can be done. At this stage though, the only testing is dpdk so I think we can keep the current scope limited to this.